Dashboard, Transaction, and Checks page and css-splitting

ia24a-germannd opened the pull request:
Consolidated and modularized CSS into common, form, animation, and other specific stylesheets. Added servlets and JSPs for dashboard, transactions, and checks to enhance functionality. Improved user session handling in servlets and applied better error handling strategies.

ia24a-germannd committed:
Split CSS files and refactor code for maintainability

Consolidated and modularized CSS into common, form, animation, and other specific stylesheets. Added servlets and JSPs for dashboard, transactions, and checks to enhance functionality. Improved user session handling in servlets and applied better error handling strategies.

ia24a-germannd committed:
Add logout button and refine UI/animations

Introduced a logout button to the dashboard for improved navigation and accessibility. Refined styles across various pages for consistency, optimized animations, and adjusted layout properties to maintain a responsive and cohesive design.

grsku commented on src/main/java/org/example/ebanking/DBConnector.java:
```diff
@@ -13,7 +11,7 @@ private Connection openConnection() {
             Class.forName("com.mysql.cj.jdbc.Driver");
             con = DriverManager.getConnection(CONNECT_STRING);
-            con.setAutoCommit(false);
+            con.setAutoCommit(true);
```
Die DB commits sollten manuell mittels `con.commit();` durchgeführt werden. 

```suggestion
            con.setAutoCommit(false);
```

grsku reviewed (changes requested):
Generell zu viel Code pro Methode (komplexität dadurch sehr hoch). Auslagern in mehrere Methoden und/oder Klassen würde dies verbessern.

ia24a-germannd committed:
Make openConnection throw SQLException and add commit call

Updated the method signature to declare SQLException, ensuring it aligns with best practices for exception handling. Removed the setAutoCommit call and added an explicit commit at the end to improve transaction control and ensure data consistency.

grsku commented on src/main/java/org/example/ebanking/DBConnector.java:
```diff
@@ -7,11 +7,10 @@ public class DBConnector {
         try {
             Class.forName("com.mysql.cj.jdbc.Driver");
             con = DriverManager.getConnection(CONNECT_STRING);
-            con.setAutoCommit(true);
```
`con.setAutoCommit(true);` muss auf false gesetzt werden und nicht entfernt. Da durch die Entfernung der Standard (true) gilt.

ia24a-germannd committed:
Set auto-commit to false and add transaction management.

This update ensures database operations are wrapped in transactions by disabling auto-commit. Commits are now explicitly triggered, and rollback is performed in case of exceptions, improving reliability and consistency in the database interactions.
